install/sct_env

```diff
@@ -8,23 +8,52 @@
 # License: see the file LICENSE.TXT
 #########################################################################################
 
-function addtopath {
+function path_prepend {
   # will not add the path again if it is already there
   # usage: addtopath "/a/directory/to/be/added" PATH_VARIABLE
-  if [[ ":${!2}:" != *":$1:"* ]]; then
-    bidon1="${1}:${!2}"
-    eval "${2}=$bidon1"
+  eval "local path_variable=$(eval echo '$'$2)"
+  local sub=":$1:"
+  if [ "${path_variable/$sub/}" = "${path_variable}" ]; then
+    if [ -z $path_variable ]; then
+      eval "$2=$1"
+    else
+      eval "$2=$1:$path_variable"
+    fi
   fi
 }
 
-SCT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+function guess_shell {
+  if [ -n "$ZSH_VERSION" ]; then
+    echo zsh
+  elif [ -n "$BASH_VERSION" ]; then
+    echo bash
+  else
+    echo "guess_shell() error: unknown shell"
+  fi
```

Comment:
I think it is important to support tcsh, a lot of sct users are on older macs.

Reply:
The thing is, master's sct_env doesn't work with tcsh...

Reply:
Conda also doesn't like tcsh (`$SCT_PYTHON/bin/activate`).

Note that I'd have liked to add dash support, but as is, it can't work, ie. I couldn't find a way to get the path of a sourced script in plain sh/dash.
For obscure shell users, it's simpler to just write `SCT_DIR` by hand (if it's ever needed).